### ISSUE TITLE
Update package-lock to apply new uuid version

### ISSUE THREAD

**Issue by XXimiCC** [optimizely/javascript-sdk#791]
## What I wanted to do
Just use sdk from box, in project which have uuid v8.3.2

## What I expected to happen
sdk works without problem

## What actually happened
I have error when I try get information about feature, there conflict uuid versions
<img width="549" alt="image" src="https://user-images.githubusercontent.com/12829885/194865214-f8e4e6e9-32e7-4ba4-9e85-47403be382f2.png">

[javascript-sdk](https://github.com/optimizely/javascript-sdk) installs old version of dependencies
<img width="440" alt="image" src="https://user-images.githubusercontent.com/12829885/194864876-505e407b-8848-42ce-8d24-547e369cf641.png">
## Steps to reproduce
Link to repository that can reproduce the issue: <link>

1. Install uuid v8.3.2 in your project,
2. Install javascript-sdk
3. Try to get feature information

**`@optimizely/optimizely-sdk` version:4.9.2** 

**Browser and version:**

**`node` version: v16.15.0** 

**`npm` version: 8.5.5** 

Versions of any other relevant tools (like module bundlers, transpilers, etc.):

**Comment by mikechu-optimizely** [comment]
Thanks for reporting @XXimiCC.  We'll start putting this through our workflow to review.

**Comment by serrg** [comment]
I confirm the issue with using `@optimizely/optimizely-sdk`  together with `uuid` v8.3.2:

```$ tsc --build
../../node_modules/@optimizely/optimizely-sdk/lib/utils/fns/index.ts:65:10 - error TS2349: This expression is not callable.
  Type 'typeof import("/node_modules/@types/uuid/index")' has no call signatures.

65   return v4()
            ~~

Found 1 error.
```

**Comment by mikechu-optimizely** [comment]
Let me look back at where updating this dep is on our project board.

cc @opti-jnguyen @zashraf1985

**Comment by mikechu-optimizely** [comment]
@serrg and @XXimiCC to be transparent: I could not find this on our dev board.  I'm unsure if it got lost in our Jira migration, but it could have also been on me not getting it onto the backlog.  In either case, I'm sorry for letting this slip by for more than a month. 

I've entered a ticket (FSSDK-8687 for ref) and will discuss during our sprint planning this week along with getting eyes on it.

Thanks again for reporting and for your patience as we move this through our process.

**Comment by mikechu-optimizely** [comment]
I presented this to the team and it looks like we need to clean up the package-lock files in both locations 
- javascript-sdk/  (likely should be removed)
- javascript-sdk/packages/optimizely-sdk/

Jira item pointed and moved into work pipeline.  

We'll keep you posted.

**Comment by connected-jzheng** [comment]
Anyone have a temporary fix for this issue? or perhaps an ETA from the team? this is holding up our codebase for adding in A/B testing since it only breaks for us when we try to add that as a rule.

**Comment by mikechu-optimizely** [comment]
The work item is slated for next week's sprint. 

Does removing the root package-lock (javascript-sdk/package-lock.json) resolve the problem?

**Comment by kvdinchev** [comment]
We have a very big project which uses uuid 8.3.4, and we are getting an error when trying to using optimizely-sdk but we are getting and `Error: node_modules/@optimizely/optimizely-sdk/lib/utils/fns/index.ts:65:10 - error TS2349: This expression is not callable.`

Are you planning on fixing this anytime soon, as it is a blocker for our infrastructure

**Comment by Tamara-Barum** [comment]
Sincere apologies for the delay. The team has an internal ticket for this work and it is ready to start. I will talk to the Product Owner and Engineer at Standup today and see how soon we can get this fix in.

**Comment by kvdinchev** [comment]
@Tamara-Barum 2 Months have passed, is the update of uuid planned for any futuristic release or it was postponed once again

Thanks in advance

**Comment by mikechu-optimizely** [comment]
Hi All, 

Thanks for hanging in there with us.  

Our primary JS dev (I'm just a padawan) has completed his research on the internal ticket (FSSDK-8687).  He should post shortly.

**Comment by raju-opti** [comment]
Hello! I am trying to reproduce this issue, but everything is working well for me. I am using `@optimizely/optimizely-sdk 4.9.2`, `uuid 8.3.2` and `@types/uuid 8.3.2`. Here is my sample repository [issue_791](https://github.com/raju-opti/issue_791). Could you please share which bundler you are using and with what configuration, or contribute a PR to my repository that reproduces the issue?

@XXimiCC @serrg @kvdinchev @connected-jzheng

**Comment by tachojelev** [comment]
We also experience the issue every time we go to production. IMO, there really is no reason for using the older version.

- New applications and frameworks use the uuid library for various reasons and most of them use the newer version.
- New versions of the uuid library are improved and provide new features.
- Optimizely is a popular library that is used in combination with the uuid library.

@raju-opti This has been an issue for quite a lot of time, is there a major show-stopper that prevent us from bumping this peer dependency and possibly improve the development experience? If not, I believe it is best to bump it, since a lot of developers are obviously having issues with this.

**Comment by kvdinchev** [comment]
This was only reproducible on Production environment, however if you dive into the uuid changes you will find out that the way uuid has to be imported in the newer version does not match the import used within the optimizely-sdk, there for it is unable to find it runtime.

Further more, the uuid is major versions apart from what is used in optimizely-sdk, keeping up with the versions mitigates potential security issues and in this case it completely changes the implementation of how a UUID is generated, as there was an issue that can cause a duplicate uuid to be generated

**Comment by georgemitwasi** [comment]
Hi @raju-opti! Providing some more context from my end so hopefully you can reproduce these issues:

This is the error I get when building my React application:
```
node_modules/@optimizely/optimizely-sdk/lib/utils/fns/index.ts:65:10 - error TS2349: This expression is not callable.
  Type 'typeof import("/Users/george.mitwasi/Repos/react-v18/services/shared/workspaces/flexcar/node_modules/@types/uuid/index")' has no call signatures.

65   return v4()
```

And here's a trimmed down version of our `package.json`:
```
{
  "scripts": {
    ...
  },
  "eslintConfig": {
    ...
  },
  "dependencies": {
    "@optimizely/react-sdk": "2.9.1",
    "@reduxjs/toolkit": "1.9.5",
    "@types/uuid": "8.3.0",
    "axios": "0.21.4",
    "prop-types": "15.7.2",
    "query-string": "6.13.7",
    "react-dom": "18.2.0",
    "react-placeholder": "4.1.0",
    "react-redux": "8.1.1",
    "react-router-dom": "6.13.0",
    "react": "18.2.0",
    "redux-persist": "6.0.0",
    "redux": "4.2.1",
    "uuid": "^8.3.2",
    "yup": "0.30.0"
  },
  "devDependencies": {
    "@babel/core": "7.12.3",
    "@babel/plugin-transform-react-jsx": "7.17.3",
    "@babel/plugin-transform-runtime": "7.17.0",
    "@babel/preset-env": "7.16.11",
    "@babel/preset-typescript": "7.16.7",
    "@babel/runtime": "7.17.9",
    "@testing-library/jest-dom": "5.16.5",
    "@testing-library/react-hooks": "8.0.1",
    "@testing-library/react": "14.0.0",
    "@testing-library/user-event": "14.4.3",
    "@types/react-dom": "18.2.5",
    "@types/react-redux": "7.1.25",
    "@types/react-router-dom": "5.3.3",
    "@types/react": "18.2.12",
    "@types/uuid": "8.3.0",
    "@types/yup": "0.29.9",
    "babel-jest": "27.5.1",
    "babel-loader": "8.2.1",
    "react-is": "18.2.0",
    "react-test-renderer": "18.2.0",
    "ts-loader": "9.4.1",
    "webpack": "5.6.0"
  }
}
```

`@optimizely/react-sdk` [v2.9.1](https://github.com/optimizely/react-sdk/blob/2.9.1/package.json#L33) has a dependency on `optimizely/optimizely-sdk` [^v4.9.1](https://github.com/optimizely/javascript-sdk/tree/v4.9.4) and this is where the issue presents itself on my end:



<img width="842" alt="Screenshot 2023-07-12 at 1 31 08 PM" src="https://github.com/optimizely/javascript-sdk/assets/17712980/a8d5215b-d085-4368-bf18-2f572f9a591f">
<img width="842" alt="Screenshot 2023-07-12 at 1 31 28 PM" src="https://github.com/optimizely/javascript-sdk/assets/17712980/4c6637e2-bd3b-4b68-9270-e4842561feb3">

Let me know if there are any other details you might need. Thanks!

**Comment by serrg** [comment]
> Hello! I am trying to reproduce this issue, but everything is working well for me. I am using `@optimizely/optimizely-sdk 4.9.2`, `uuid 8.3.2` and `@types/uuid 8.3.2`. Here is my sample repository [issue_791](https://github.com/raju-opti/issue_791). Could you please share which bundler you are using and with what configuration, or contribute a PR to my repository that reproduces the issue?
> 
> @XXimiCC @serrg @kvdinchev @connected-jzheng

When you are using `@types/uuid 8.3.2` version, types are working good but in the `@optimizely/optimizely-sdk`  dependencies there is still used `3.4.4` version: `"@types/uuid": "^3.4.4",` 

Link here: https://github.com/optimizely/javascript-sdk/blob/master/packages/optimizely-sdk/package.json#L59

**Comment by Tamara-Barum** [comment]
FSSDK-8687

**Comment by raju-opti** [comment]
@optimizely/optimizely-sdk version 4 supports IE 10. uuid versions greater than 3 does not support IE 10. So upgrading uuid version for optimizely-sdk version 4 will be a breaking change. We have updated uuid to version 8 for optimizely-sdk version 5 which is currently in beta (https://www.npmjs.com/package/@optimizely/optimizely-sdk/v/5.0.0-beta5).

**Comment by nrj** [comment]
This issue seems to be fixed for me in [5.0.0-beta5](https://www.npmjs.com/package/@optimizely/optimizely-sdk/v/5.0.0-beta5) - is there a timeline for this to be considered production ready?

**Comment by russell-loube-optimizely** [comment]
Hey @nrj , unfortunately we do not yet have a timeline for exiting beta. We have certain beta exit criteria for our newest Advanced Audience Targeting functionality that must be met before promoting to GA.

I'm sorry not to have a better answer for you at the moment.

**Comment by raju-opti** [comment]
Version 5.0.0 has been released with uuid@9.0.1. Closing this issue.
